src/lib/mdbm_lock.cc

```diff
@@ -352,7 +352,10 @@ struct mdbm_locks* open_locks_inner(const char* dbname, int flags, int do_lock,
     count = get_cpu_count() * 2;
   } else if (flags & MDBM_PARTITIONED_LOCKS) {
     type = MLOCK_INDEX;
-    count = 128;
+    count = get_cpu_count() * 16;
+    if (count < 128) {
+      count = 128;
+    }
```

Comment:
minor thing, but maybe it could look a bit cleaner if it was:
`count = max(128, get_cpu_count() * 16)`